### PR TITLE
Update test files to load assets via GUID instead of path

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview

In preparation for #7409, this updates all tests to load assets via GUID instead of hardcoded path. This will make our tests more robust to folder / path changes in the future, while also potentially alerting us to GUID changes that we'll need to mitigate.

## Changes
- Fixes: #7438

### CODE REVIEW

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/Utils/TestButtonUtilities.cs [hunk 1]
```diff
@@ -30,27 +30,27 @@ public enum DefaultButtonType
...
         /// </summary>
-        public const string PressableHoloLens2PrefabPath = "Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Prefabs/PressableButtonHoloLens2.prefab";
+        public static readonly string PressableHoloLens2PrefabPath = AssetDatabase.GUIDToAssetPath(PressableHoloLens2PrefabGuid);
 
         /// <summary>
-        /// Asset path to default Hololens 2 Toggle Pressable Button prefab asset
+        /// Asset path to default HoloLens 2 Toggle Pressable Button prefab asset
         /// </summary>
-        public const string PressableHoloLens2TogglePrefabPath = "Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Prefabs/PressableButtonHoloLens2Toggle.prefab";
+        public static readonly string PressableHoloLens2TogglePrefabPath = AssetDatabase.GUIDToAssetPath(PressableHoloLens2TogglePrefabGuid);
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/Utils/TestButtonUtilities.cs]: Realized there's a less invasive way to do this, which would be to add variables of this nature in each test class I updated. I'm gonna move to this pattern before merging

File: Assets/MixedRealityToolkit.Tests/EditModeTests/InputSystem/InputSystemTests.cs [hunk 2]
```diff
@@ -12,8 +12,13 @@ namespace Microsoft.MixedReality.Toolkit.Tests.EditMode.InputSystem
...
     public class InputSystemTests
     {
-        private const string TestInputSystemProfilePath = "Assets/MixedRealityToolkit.Tests/EditModeTests/Services/TestProfiles/TestMixedRealityInputSystemProfile.asset";
-        private const string TestEmptyInputSystemProfilePath = "Assets/MixedRealityToolkit.Tests/EditModeTests/Services/TestProfiles/TestEmptyMixedRealityInputSystemProfile.asset";
+        // Tests/EditModeTests/Services/TestProfiles/TestMixedRealityInputSystemProfile.asset
+        private const string TestInputSystemProfileGuid = "9c5be3fea81382342b84623b5b225012";
+        private static readonly string TestInputSystemProfilePath = AssetDatabase.GUIDToAssetPath(TestInputSystemProfileGuid);
+
+        // Tests/EditModeTests/Services/TestProfiles/TestEmptyMixedRealityInputSystemProfile.asset
+        private const string TestEmptyInputSystemProfileGuid = "45c96c8b3073919419eae31f72ca8e4e";
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit.Tests/EditModeTests/InputSystem/InputSystemTests.cs]: Would it be easier to use the MRTK files lookup? This feels like adding steps (open meta file, find guid, etc) for developers.

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Tests/EditModeTests/InputSystem/InputSystemTests.cs]: on the other hand, is this a faster access method for test collateral (ex: no file system lookup)

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Tests/EditModeTests/InputSystem/InputSystemTests.cs]: My thought was that the MRTKFiles APIs seemed to do a lot regarding searching directories and everything. Depending on how Unity maps GUIDs to paths, this felt like a more performant solution, and one that would be more robust to file name or folder name changes.

I can run some tests today to see perf between the two.

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Tests/EditModeTests/InputSystem/InputSystemTests.cs]: Ran the following test. Let me know if it looks like I misconfigured anything, but the results were 

image


```c#
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

using Microsoft.MixedReality.Toolkit.Input;
using Microsoft.MixedReality.Toolkit.SpatialAwareness;
using Microsoft.MixedReality.Toolkit.Utilities.Editor;
using System.Diagnostics;
using UnityEditor;
using UnityEngine;

public class TestFileLoading : MonoBehaviour
{
    private const string TestInputSystemProfileGuid = "9c5be3fea81382342b84623b5b225012";
    private const string TestEmptyInputSystemProfileGuid = "45c96c8b3073919419eae31f72ca8e4e";
    private const string TestSpatialAwarenessSystemProfileGuid = "757d72d70f9c56144b05346288682e00";
    private const string CameraSystemPrefabGuid = "8a5d96f565aaee14ca3e07575bb9ce80";
    private const string RadialSetPrefabAssetGuid = "8b83134143223104c9bc3865a565cab3";

    private const string TestInputSystemProfilePath = "EditModeTests/Services/TestProfiles/TestMixedRealityInputSystemProfile.asset";
    private const string TestEmptyInputSystemProfilePath = "EditModeTests/Services/TestProfiles/TestEmptyMixedRealityInputSystemProfile.asset";
    private const string TestSpatialAwarenessSystemProfilePath = "EditModeTests/Services/TestProfiles/TestMixedRealitySpatialAwarenessSystemProfile.asset";
    private const string CameraSystemPrefabPath = "Experimental/ServiceManagers/Camera/Prefabs/CameraSystem.prefab";
    private const string RadialSetPrefabAssetPath = "Features/UX/Interactable/Prefabs/RadialSet.prefab";

    private void Start()
    {
        Stopwatch stopwatch = Stopwatch.StartNew();

        for (int i = 0; i < 1000; i++)
        {
            _ = AssetDatabase.LoadAssetAtPath<MixedRealityInputSystemProfile>(AssetDatabase.GUIDToAssetPath(TestInputSystemProfileGuid));
            _ = AssetDatabase.LoadAssetAtPath<MixedRealityInputSystemProfile>(AssetDatabase.GUIDToAssetPath(TestEmptyInputSystemProfileGuid));
            _ = AssetDatabase.LoadAssetAtPath<MixedRealitySpatialAwarenessSystemProfile>(AssetDatabase.GUIDToAssetPath(TestSpatialAwarenessSystemProfileGuid));
            _ = AssetDatabase.LoadAssetAtPath<Object>(AssetDatabase.GUIDToAssetPath(CameraSystemPrefabGuid));
            _ = AssetDatabase.LoadAssetAtPath<Object>(AssetDatabase.GUIDToAssetPath(RadialSetPrefabAssetGuid));
        }

        stopwatch.Stop();

        UnityEngine.Debug.Log($"Load by GUID took {stopwatch.ElapsedMilliseconds} milliseconds.");

        stopwatch.Reset();
        stopwatch.Start();

        for (int i = 0; i < 1000; i++)
        {
            _ = AssetDatabase.LoadAssetAtPath<MixedRealityInputSystemProfile>(MixedRealityToolkitFiles.MapRelativeFilePath(MixedRealityToolkitModuleType.Tests, TestInputSystemProfilePath));
            _ = AssetDatabase.LoadAssetAtPath<MixedRealityInputSystemProfile>(MixedRealityToolkitFiles.MapRelativeFilePath(MixedRealityToolkitModuleType.Tests, TestEmptyInputSystemProfilePath));
            _ = AssetDatabase.LoadAssetAtPath<MixedRealitySpatialAwarenessSystemProfile>(MixedRealityToolkitFiles.MapRelativeFilePath(MixedRealityToolkitModuleType.Tests, TestSpatialAwarenessSystemProfilePath));
            _ = AssetDatabase.LoadAssetAtPath<Object>(MixedRealityToolkitFiles.MapRelativeFilePath(MixedRealityToolkitModuleType.SDK, CameraSystemPrefabPath));
            _ = AssetDatabase.LoadAssetAtPath<Object>(MixedRealityToolkitFiles.MapRelativeFilePath(MixedRealityToolkitModuleType.SDK, RadialSetPrefabAssetPath));
        }

        stopwatch.Stop();

        UnityEngine.Debug.Log($"Load by MRTKFiles took {stopwatch.ElapsedMilliseconds} milliseconds.");
    }
}
```

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Tests/EditModeTests/InputSystem/InputSystemTests.cs]: It is negligible in our scenario, but very enlightening. I'm good with using GUIDs provided that we always use them to access test collateral :)

**Comment by davidkline-ms** [comment]
:shipit:

**Pull request merged**